java.hints/src/org/netbeans/modules/java/hints/suggestions/ConvertToVarHint.java

```diff
@@ -0,0 +1,449 @@
+import org.netbeans.spi.java.hints.HintContext;
+import org.netbeans.spi.java.hints.JavaFix;
+import org.netbeans.spi.java.hints.JavaFix.TransformationContext;
+import org.netbeans.spi.java.hints.TriggerPattern;
+import org.openide.util.NbBundle.Messages;
+import org.netbeans.spi.editor.hints.Fix;
+import org.openide.filesystems.FileObject;
+import org.openide.modules.SpecificationVersion;
+import org.netbeans.spi.java.hints.TriggerTreeKind;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Hint will convert explicit type to 'var'. Supported: JDK 10 or above
+ *
+ * @author arusinha
+ */
+public class ConvertToVarHint {
+
+    private static final SpecificationVersion JDK_10 = new SpecificationVersion("10"); //NOI18N
+
+    private final static Set<Tree.Kind> LITERALS_TYPE_SET = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    Tree.Kind.INT_LITERAL,
+                    Tree.Kind.LONG_LITERAL,
+                    Tree.Kind.FLOAT_LITERAL,
+                    Tree.Kind.DOUBLE_LITERAL,
+                    Tree.Kind.BOOLEAN_LITERAL,
+                    Tree.Kind.CHAR_LITERAL,
+                    Tree.Kind.STRING_LITERAL
+            )));
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return ErrorDescription in case Object reference type and Object
+     * instance type are same.
+     */
+    @Hint(displayName = "#DN_CanUseVarForObjectRef", description = "#DESC_CanUseVarForObjectRef", category = "suggestions") //NOI18N
+
+    @TriggerPattern("$mods$ $type $var =new $type($params$);") //NOI18N
+    @Messages("MSG_ConvertibleToVarType=Explict type can be replaced with 'var'")  //NOI18N
+    public static ErrorDescription checkNewObjInit(HintContext ctx) {
+
+        TreePath treePath = ctx.getVariables().get("$var");
+
+        if (treePath == null) {
+            return null;
+        }
+
+        if (!isHintEnabled(ctx)) {
+            return null;
+        }
+
+        Fix fix = new FixImpl(ctx.getInfo(), treePath).toEditorFix();
+        return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.MSG_ConvertibleToVarType(), fix);
+
+    }
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return ErrorDescription if variable initializer is a Literal
+     */
+    @Hint(displayName = "#DN_CanUseVarForLiteralRef", description = "#DESC_CanUseVarForLiteralRef", category = "suggestions")
+    @TriggerPattern("$mods$ $type $var = $value;") //NOI18N
+
+    public static ErrorDescription checkLiteralInit(HintContext ctx) {
+
+        TreePath initPath = ctx.getVariables().get("$value");
+
+        if (initPath == null) {
+            return null;
+        }
+
+        Tree l = initPath.getLeaf();
+
+        // checks rhs is a literal or not
+        if (!(isLiteral(l))) {
+            return null;
+        }
+
+        if (!isHintEnabled(ctx)) {
```

Comment:
Same here why is isHintEnabled checked so late?

Reply:
https://github.com/apache/incubator-netbeans/pull/463/commits/b733846a4ab3081e2b4d56531ab3c3d041fdd93e